js/main.js

```diff
@@ -1,44 +1,121 @@
+  var $notes = document.querySelector('.textbox').value;
+  var entryData = {
+    title: $title,
+    photoURL: $photoURL,
+    notes: $notes
   };
-  entryData.entryId = data.nextEntryId;
+  entryData.entryId = data.nextEntryId; // creating a new key for entryData
+  entryList.prepend(userEntry(entryData));
   data.nextEntryId++; // incrementing
-  data.entries.unshift(entryData); // prepends. prepends means to put it in the beginning of something which is used as a .unshift();. With each data SAVED, it will be placed in the beginning of the object.
-  // localStorage.setItem('New Data', JSON.stringify(entryData)); // storing in localstorage by using localStorage.setItem(). Must use JSON.stringify() because localStorage only accepts data as a string.
+  data.entries.unshift(entryData);
+  localStorage.setItem('JavaScriptLocalStorage', JSON.stringify({ ...data, entries: [entryData, ...oldEntries.entries] })); // entries: [entryData, ...oldEntries.entries] == reassigning entries to an array with all of the entries of the oldEntries and adding onto entryData
   $imgSrc.setAttribute('src', './images/placeholder-image-square.jpg');
   $entryForm.reset(); // resets the entryForm once user clicks SAVE button
 }
 
-window.addEventListener('beforeunload', function (event) {
-  var stringData = JSON.stringify(data);
-  localStorage.setItem('New data', stringData);
-});
-var previousData = localStorage.getItem('javascript-local-storage');
-if (previousData !== null) {
-  data.parse(previousData);
+// make a function that will save user entries after pressing the save BUTTON.
+/* $saveButton.addEventListener('click', function (event) {
+  var userDataEntry = {
+    title: $entryForm.elements.title.value,
+    photoURL: $photoUrl.value,
+    notes: $entryForm.elements.notes.value
+  };
+  data.entries.unshift(userDataEntry);
+}); */
+
+// create a DOM tree for our html sample journal entry that will change/update with JavaScript.
+function userEntry(entryObject) {
+  var $liElement = document.createElement('li');
+  $liElement.setAttribute('class', 'row');
+
+  var $divFirstColumn = document.createElement('div');
+  $divFirstColumn.setAttribute('class', 'column-half');
+  $liElement.appendChild($divFirstColumn);
+
+  var $img = document.createElement('img');
+  $img.setAttribute('src', entryObject.photoURL);
+  $divFirstColumn.appendChild($img);
+
+  var $divSecondColumn = document.createElement('class', 'column-half');
+  $liElement.appendChild($divSecondColumn);
+
+  var $hThree = document.createElement('h3');
+  $hThree.textContent = entryObject.title;
+  $divSecondColumn.appendChild($hThree);
+
+  var $pContent = document.createElement('p');
+  $pContent.textContent = entryObject.notes;
+  $divSecondColumn.appendChild($pContent);
+  return $liElement;
 }
+
+function DOMContentLoaded() {
+  const entries = localStorage.getItem('JavaScriptLocalStorage');
+  const previousData = JSON.parse(entries);
```

Comment:
This shouldn't be necessary, you should be parsing what is in localStorage right when the page loads in the `data.js` file. You should review `javascript-local-storage`, the localStorage-related code for this project will be **_very similar_** to the code you wrote for the exercise.